dumb-jump.el

```diff
@@ -2067,27 +2073,101 @@ to keep looking for another root."
+    (if lang
+        (dumb-jump-make-composite-language
+         "org"
+         (if (dumb-jump-get-language-from-aliases lang)
+             (dumb-jump-get-language-from-aliases lang) lang)
+         "org" "org" "org")
+      "org")))
+
+(defun dumb-jump-add-language-to-proplist (newlang proplist lang)
+  "Return nil if NEWLANG  is already in PROPLIST or (if not)
+return a new proplist. The new proplis is PROPLIS
+where a NEWLANG plist(s) is (are) added to PROPLIST.
+The plist(s) value of NEWLANG is (are) copied from
+those of LANG and LANG is replaced by NEWLANG." 
+  (let ((alreadyuptodate
+         (--filter (string= newlang (plist-get it :language))
+                   proplist)))
+    (if alreadyuptodate
+        nil
+      (--splice
+       (string= lang (plist-get it :language))
+       (list it (plist-put (copy-tree it) :language newlang))
+       proplist))))
+
+(defun dumb-jump-make-composite-language (mode lang extension agtype rgtype)
+  "Concat one MODE  (usually the string org) with a LANG  (c or python or etc)
+to make a composite language of the form cPLUSorg or pythonPLUSorg or etc. 
+Modify `dumb-jump-find-rules' and `dumb-jump-language-file-exts' accordingly
+(using EXTENSION AGTYPE RGTYPE)"
+  (let* ((complang (concat lang "PLUS" mode))
+         (alreadyextension (--filter (and
+                                      (string= complang (plist-get it :language))
+                                      (string= extension (plist-get it :ext))
+                                      (string= agtype (plist-get it :agtype))
+                                      (string= rgtype (plist-get it :rgtype)))
+                                     dumb-jump-language-file-exts))
+         (newfindrule (dumb-jump-add-language-to-proplist complang dumb-jump-find-rules lang))
+         (newfileexts (dumb-jump-add-language-to-proplist complang dumb-jump-language-file-exts lang)))
+    ;; add (if needed) composite language to dumb-jump-find-rules
+    (if newfindrule
+        (set-default 'dumb-jump-find-rules newfindrule))
+    ;; add (if needed) composite language to dumb-jump-language-file-exts
+    (if (not dumb-jump-search-type-org-only-org)
+        (if newfileexts
+            (set-default 'dumb-jump-language-file-exts newfileexts)))
+    ;; add (if needed) a new extension to dumb-jump-language-file-exts
+    (if (not alreadyextension)
+        (set-default 'dumb-jump-language-file-exts
+                     (cons `(:language ,complang :ext ,extension :agtype ,agtype :rgtype ,rgtype)
+                           dumb-jump-language-file-exts)))
+    complang))
+
+(defun dumb-jump-get-language-from-aliases (lang)
+  "Extract the lang from aliases."
+  (let* ((lookup '(sh "shell" shell "shell" cperl "perl"
+                      matlab "matlab" octave "matlab"
+                      emacs-lisp "elisp" elisp "elisp"
+                      R "r" r "r"))
+         (result (plist-get lookup (intern lang))))
+    (if result result nil)))
```

Comment:
The if is redundant, but I'd also change the entire function to use assoc/alist-get or cond, instead of plist-get.

```
(alist-get lang '(("sh" . "shell") ("shell" . "shell") ...))
```